Tests/CommandsTests/PackageToolTests.swift

```diff
@@ -1541,4 +1541,130 @@ final class PackageToolTests: CommandsTestCase {
             }
         }
     }
+
+    func testCommandPluginTestingCallbacks() throws {
+        // Depending on how the test is running, the `llvm-profdata` and `llvm-cov` tool might be unavailable.
+        try XCTSkipIf((try? UserToolchain.default.getLLVMProf()) == nil, "skipping test because the `llvm-profdata` tool isn't available")
+        try XCTSkipIf((try? UserToolchain.default.getLLVMCov()) == nil, "skipping test because the `llvm-cov` tool isn't available")
+
+        try testWithTemporaryDirectory { tmpPath in
+            // Create a sample package with a library, a command plugin, and a couple of tests.
+            let packageDir = tmpPath.appending(components: "MyPackage")
+            try localFileSystem.createDirectory(packageDir, recursive: true)
```

Comment:
We should have a version of `writeFileContents(_:string:)` that can create intermediate directories when needed.

In the long run it would be useful for unit tests to have a way of creating a file system using something like result builders, to lay out contents in the file system like SwiftUI lays out views on the screen.  We could then have a library of generator for common kinds of files.  For now, using the string versions of `writeFileContents` seems better than the stream block version even though we have to create the directories ourselves.

Reply:
one interesting thing I found was that `writeFileContents(_ path: AbsolutePath, body: (WritableByteStream) -> Void) throws` creates the directory while `func writeFileContents(_ path: AbsolutePath, bytes: ByteString)`. its a subtle difference and not super clear to me why one does it over the other, but we should probably may the behavior the same across the board, maybe by adding an optional argument `createDirectory` with no default such that the test code can create an extension that sets it to true. wdyt?

Reply:
cc @neonichu

Reply:
Right, that's exactly what I had in mind (the extra parameter).

Reply:
Since this PR uses the string forms I don't think we should block it on changing `writeFileContents` though.

Reply:
probably a radar to track the desired refactoring?

Reply:
Definitely.  I'll file a bug on bugs.swift.org.

Reply:
Filed https://bugs.swift.org/browse/SR-15719